shared/src/pool_fetching.rs

```diff
@@ -189,30 +208,47 @@ impl CachedPoolFetcher {
         at_block: BlockNumber,
     ) -> Vec<Pool> {
         let mut cache = self.cache.lock().await;
-        //TODO check that cache hit also matches at_block parameter
+        let block = match at_block {
+            BlockNumber::Earliest => 0,
+            BlockNumber::Number(number) => number.as_u64(),
+            BlockNumber::Latest => cache.latest_block_number(),
+            BlockNumber::Pending => {
+                tracing::warn!("Pending block not supported by cache");
+                return self.inner.fetch(token_pairs, at_block).await;
+            }
+        };
+        let mut cached_pools = cache.pools.pop(&block).unwrap_or_else(HashMap::new);
+
         let (cache_hits, cache_misses) = token_pairs
             .into_iter()
-            .partition::<HashSet<_>, _>(|pair| cache.1.contains_key(pair));
+            .partition::<HashSet<_>, _>(|pair| cached_pools.contains_key(pair));
         let cache_results: Vec<_> = cache_hits
             .iter()
-            .filter_map(|pair| cache.1.get(pair))
+            .filter_map(|pair| cached_pools.get(pair))
             .flatten()
             .cloned()
             .collect();
 
         let mut inner_results = self.inner.fetch(cache_misses, at_block).await;
         for miss in &inner_results {
-            cache.1.entry(miss.tokens).or_default().push(miss.clone());
+            cached_pools
+                .entry(miss.tokens)
+                .or_default()
+                .push(miss.clone());
```

Comment:
if the entry is a vector would this be equivalent to 

```
cached_pools.entry(miss.tokens).or_default().extend(inner_results)
```

Reply:
I think so, yes.